bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -341,29 +396,25 @@ protected void updatePowerLimitationStatus(Channel channel, boolean status, int
         updateState(channel.getUID(), status ? OnOffType.ON : OnOffType.OFF);
     }
 
-    protected void updateBatteryState(Channel channel, int code) {
-        updateState(channel.getUID(), new StringType(SenecBatteryStatus.descriptionFromCode(code)));
-    }
-
     protected void updateGridPowerValues(BigDecimal gridTotalValue) {
         BigDecimal gridTotal = gridTotalValue.setScale(2, RoundingMode.HALF_UP);
 
         Channel channelGridPower = getThing().getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_POWER);
         if (channelGridPower != null) {
-            updateState(channelGridPower.getUID(), new QuantityType<Power>(gridTotal, Units.WATT));
+            updateState(channelGridPower.getUID(), new QuantityType<>(gridTotal, Units.WATT));
```

Comment:
Why was the <Power> removed?

Reply:
Java type inference: the compiler can "gues" the correct type from generics definition (because WATT is defined as Unit<Power>).
This "feature" is also used on the various "updateQtyState" methods I introduced. The QuantityType infers its generic type from the "unit".